src/main/java/uk/gov/hmcts/ccd/domain/service/startevent/DefaultStartEventOperation.java

```diff
@@ -115,31 +136,30 @@ public StartEventTrigger triggerStartForDraft(final String uid,
                                                   final String draftReference,
                                                   final String eventTriggerId,
                                                   final Boolean ignoreWarning) {
-        return buildStartEventTrigger(uid,
-                                      jurisdictionId,
-                                      caseTypeId,
+        final CaseType caseType = getCaseType(caseTypeId);
+
+        return buildStartEventTrigger(() -> uid,
+                                      caseType,
                                       eventTriggerId,
                                       ignoreWarning,
-            () -> getDraftDetails(draftReference));
+                                      () -> getDraftDetails(draftReference));
     }
 
-    private StartEventTrigger buildStartEventTrigger(final String uid,
-                                                     final String jurisdictionId,
-                                                     final String caseTypeId,
+    private StartEventTrigger buildStartEventTrigger(final Supplier<String> uidSupplier,
```

Comment:
Why is a supplier needed for the `uid`?

Reply:
Leftover from previous approach. Will cleanup.